paths/src/main/java/coursier/CoursierPaths.java

```diff
@@ -27,19 +27,14 @@ public static File cacheDirectory() {
         if (path == null) {
         	File coursierDotFile = new File(System.getProperty("user.home") + "/.coursier");
         	if (coursierDotFile.isDirectory())
-        		path = System.getProperty("user.home") + "/.coursier/cache/";
+                path = System.getProperty("user.home") + "/.coursier/cache/v1/";
         }
 
         if (path == null) {
         	 coursierDirectories = ProjectDirectories.fromProjectName("Coursier");
         	 path = coursierDirectories.projectCacheDir;
         }
 
-        File coursierCacheDirectory = new File(path).getAbsoluteFile();
-
-        if (coursierCacheDirectory.getName().equals("v1"))
-        	coursierCacheDirectory = coursierCacheDirectory.getParentFile();
-        
-        return coursierCacheDirectory.getAbsoluteFile();
```

Comment:
I made these changes to separate figuring out the various directories in `CoursierPaths` from the way they are used `CachePath`.

This way dealing with "v1" makes it easy for Coursier to potentially upgrade to a "v2" layout later, without breaking everyone's environment settings pointing to the "v1" dir.

The idea is that you only need to adapt `defaultCacheDirectory` to use "v2" and be done without having to write migration code and documenting how users need to change their environment variables.